packages/host/app/services/command-service.ts

```diff
@@ -0,0 +1,68 @@
+import Service from '@ember/service';
+
+type Command = 'patch';
+
+class CommandHandler {
+  handler: any;
+  functions: Map<Command, ((arg: any) => void)[]>;
+  constructor(handler: any) {
+    this.handler = handler;
+    this.functions = new Map();
+  }
+
+  public get(command: Command) {
+    if (!this.functions.has(command)) {
+      this.functions.set(command, []);
+    }
+    return this.functions.get(command);
+  }
+
+  public async handle(command: Command, arg: any) {
+    if (!this.functions.has(command)) {
+      throw new Error(`No handler registered for command ${command}`);
+    }
+    const functions = this.functions.get(command)!;
```

Comment:
if the `.get()` side effect what to make the for loop happy, then there is a much more elegant way of doing that:
```
for (const func of functions ?? []) {
``` 
let's do that instead of having side effects in our getters